pkg/pgmodel/querier/query_builder.go

```diff
@@ -33,11 +35,19 @@ const (
 	subQueryREMatchEmpty  = "NOT labels && (SELECT COALESCE(array_agg(l.id), array[]::int[]) FROM _prom_catalog.label l WHERE l.key = $%d and l.value !~ $%d)"
 	subQueryNRE           = "labels && (SELECT COALESCE(array_agg(l.id), array[]::int[]) FROM _prom_catalog.label l WHERE l.key = $%d and l.value !~ $%d)"
 	subQueryNREMatchEmpty = "NOT labels && (SELECT COALESCE(array_agg(l.id), array[]::int[]) FROM _prom_catalog.label l WHERE l.key = $%d and l.value ~ $%d)"
+
+	// RE2 regex matching sub-queries using custom regex matching function.
+	subQueryRE2            = "labels && (SELECT COALESCE(array_agg(l.id), array[]::int[]) FROM _prom_catalog.label l WHERE l.key = $%d and _prom_ext.re2_match(l.value, $%d))"
+	subQueryRE2MatchEmpty  = "NOT labels && (SELECT COALESCE(array_agg(l.id), array[]::int[]) FROM _prom_catalog.label l WHERE l.key = $%d and not _prom_ext.re2_match(l.value, $%d))"
+	subQueryNRE2           = "labels && (SELECT COALESCE(array_agg(l.id), array[]::int[]) FROM _prom_catalog.label l WHERE l.key = $%d and not _prom_ext.re2_match(l.value, $%d))"
+	subQueryNRE2MatchEmpty = "NOT labels && (SELECT COALESCE(array_agg(l.id), array[]::int[]) FROM _prom_catalog.label l WHERE l.key = $%d and _prom_ext.re2_match(l.value, $%d))"
 )
 
 var (
 	minTime = timestamp.FromTime(time.Unix(math.MinInt64/1000+62135596801, 0).UTC())
 	maxTime = timestamp.FromTime(time.Unix(math.MaxInt64/1000-62135596801, 999999999).UTC())
+
+	re2Regex = regexp.MustCompile(`\(\?`)
```

Comment:
I would appreciate a comment here on why this marker is useful to identify a regex that should be treated as RE2.

Reply:
Right, will add.